src/GraphQL.Tests/Utilities/Visitors/DeprecatedTypeReferenceVisitorTests.cs

```diff
@@ -0,0 +1,273 @@
+            .Type(deprecatedType)
+            .DeprecationReason("Use newAuthor field instead");
+
+        var schema = new Schema
+        {
+            Query = activeType
+        };
+        schema.RegisterType(deprecatedType);
+        schema.RegisterVisitor(new DeprecatedTypeReferenceVisitor());
+
+        // Act & Assert
+        schema.Initialize();
+    }
+
+    [Fact]
+    public void Should_Throw_AggregateException_When_Multiple_Violations()
+    {
+        // Arrange
+        var deprecatedType1 = new ObjectGraphType
+        {
+            Name = "DeprecatedUser",
+            DeprecationReason = "Use NewUser instead"
+        };
+        deprecatedType1.Field<StringGraphType>("id");
+
+        var deprecatedType2 = new ObjectGraphType
+        {
+            Name = "DeprecatedCategory",
+            DeprecationReason = "Use NewCategory instead"
+        };
+        deprecatedType2.Field<StringGraphType>("name");
+
+        var activeType = new ObjectGraphType { Name = "Post" };
+        activeType.Field<StringGraphType>("title");
+        activeType.Field<ObjectGraphType>("author").Type(deprecatedType1);
+        activeType.Field<ObjectGraphType>("category").Type(deprecatedType2);
+
+        var schema = new Schema
+        {
+            Query = activeType
+        };
+        schema.RegisterType(deprecatedType1);
+        schema.RegisterType(deprecatedType2);
+        schema.RegisterVisitor(new DeprecatedTypeReferenceVisitor());
+
+        // Act & Assert
+        var exception = Should.Throw<AggregateException>(() => schema.Initialize());
+        exception.Message.ShouldContain("Schema validation failed. Found multiple non-deprecated fields referencing deprecated types.");
+        exception.InnerExceptions.Count.ShouldBe(2);
+
+        var innerMessages = exception.InnerExceptions.Select(ex => ex.Message).ToArray();
+        innerMessages.ShouldContain("Non-deprecated field 'Post.author' references deprecated type 'DeprecatedUser'.");
+        innerMessages.ShouldContain("Non-deprecated field 'Post.category' references deprecated type 'DeprecatedCategory'.");
+    }
+
+    [Fact]
+    public void Should_Not_Throw_When_NonDeprecated_Field_References_NonDeprecated_Type()
+    {
+        // Arrange
+        var activeUserType = new ObjectGraphType { Name = "User" };
+        activeUserType.Field<StringGraphType>("id");
+
+        var activePostType = new ObjectGraphType { Name = "Post" };
+        activePostType.Field<StringGraphType>("title");
+        activePostType.Field<ObjectGraphType>("author").Type(activeUserType);
+
+        var schema = new Schema
+        {
+            Query = activePostType
+        };
+        schema.RegisterType(activeUserType);
+        schema.RegisterVisitor(new DeprecatedTypeReferenceVisitor());
+
+        // Act & Assert
+        schema.Initialize();
+    }
+
+    [Fact]
+    public void Should_Throw_When_ObjectField_References_Deprecated_Type()
```

Comment:
Duplicate of `Should_Throw_When_NonDeprecated_Field_References_Deprecated_Type` (except `title` field)

Reply:
removed duplicate